physics/barycentric_rotating_dynamic_frame_body.hpp

```diff
@@ -0,0 +1,113 @@
+    : ephemeris_(ephemeris),
+      primary_(primary),
+      secondary_(secondary),
+      primary_trajectory_(ephemeris_->trajectory(primary_)),
+      secondary_trajectory_(ephemeris_->trajectory(secondary_)) {}
+
+template<typename InertialFrame, typename ThisFrame>
+RigidMotion<InertialFrame, ThisFrame>
+BarycentricRotatingDynamicFrame<InertialFrame, ThisFrame>::ToThisFrameAtTime(
+    Instant const& t) const {
+  DegreesOfFreedom<InertialFrame> const primary_degrees_of_freedom =
+      primary_trajectory_->EvaluateDegreesOfFreedom(t, &primary_hint_);
+  DegreesOfFreedom<InertialFrame> const secondary_degrees_of_freedom =
+      secondary_trajectory_->EvaluateDegreesOfFreedom(t, &secondary_hint_);
+  DegreesOfFreedom<InertialFrame> const barycentre_degrees_of_freedom =
+      Barycentre<InertialFrame, GravitationalParameter>(
+          {primary_degrees_of_freedom,
+           secondary_degrees_of_freedom},
+          {primary_->gravitational_parameter(),
+           secondary_->gravitational_parameter()});
+
+  Rotation<InertialFrame, ThisFrame>
+      from_basis_of_inertial_frame_to_basis_of_this_frame =
+          Rotation<InertialFrame, ThisFrame>::Identity();
+  AngularVelocity<InertialFrame> angular_velocity;
+  FromBasisOfInertialFrameToBasisOfThisFrame(
+      barycentre_degrees_of_freedom,
+      primary_degrees_of_freedom,
+      secondary_degrees_of_freedom,
+      &from_basis_of_inertial_frame_to_basis_of_this_frame,
+      &angular_velocity);
+
+  RigidTransformation<InertialFrame, ThisFrame> const
+      rigid_transformation(
+          barycentre_degrees_of_freedom.position(),
+          ThisFrame::origin,
+          from_basis_of_inertial_frame_to_basis_of_this_frame.Forget());
+  return RigidMotion<InertialFrame, ThisFrame>(
+             rigid_transformation,
+             angular_velocity,
+             barycentre_degrees_of_freedom.velocity());
+}
+
+template<typename InertialFrame, typename ThisFrame>
+RigidMotion<ThisFrame, InertialFrame>
+BarycentricRotatingDynamicFrame<InertialFrame, ThisFrame>::
+FromThisFrameAtTime(Instant const& t) const {
+  return ToThisFrameAtTime(t).Inverse();
+}
+
+template<typename InertialFrame, typename ThisFrame>
+Vector<Acceleration, ThisFrame>
+BarycentricRotatingDynamicFrame<InertialFrame, ThisFrame>::
+GeometricAcceleration(
+    Instant const& t,
+    DegreesOfFreedom<ThisFrame> const& degrees_of_freedom) const {
+  return Vector<Acceleration, ThisFrame>();
+}
+
+template<typename InertialFrame, typename ThisFrame>
+void BarycentricRotatingDynamicFrame<InertialFrame, ThisFrame>::
+FromBasisOfInertialFrameToBasisOfThisFrame(
+    DegreesOfFreedom<InertialFrame> const& barycentre_degrees_of_freedom,
+    DegreesOfFreedom<InertialFrame> const& primary_degrees_of_freedom,
+    DegreesOfFreedom<InertialFrame> const& secondary_degrees_of_freedom,
+    not_null<Rotation<InertialFrame, ThisFrame>*> const rotation,
+    not_null<AngularVelocity<InertialFrame>*> const angular_velocity) {
```

Comment:
We are not using `secondary_degrees_of_freedom` here, and in the implementation we could replace `barycentre_degrees_of_freedom` by `secondary_degrees_of_freedom` (assuming the contract `|barycentre_degrees_of_freedom| must be a convex combination of the two other degrees of freedom` is respected).

This is called only once, so it would be unneeded encapsulation, but since we want to add the 2-body corotating frame fixing one of the bodies (rather than the barycentre) in Brouwer, it could be reused there. Move this function out of the class instead, drop the `barycentre_degrees_of_freedom` parameter.

The name suggests that it returns some sort of linear map, change to `ComputeTwoBodyRotationalMotion` or something like that.